public/app/features/plugins/datasource_srv.ts

```diff
@@ -115,9 +117,12 @@ export class DatasourceSrv implements DataSourceService {
       return Promise.resolve(expressionDatasource);
     }
 
-    const dsConfig = this.settingsMapByName[name];
+    let dsConfig = this.settingsMapByName[name];
     if (!dsConfig) {
-      return Promise.reject({ message: `Datasource named ${name} was not found` });
+      dsConfig = this.settingsMapById[name];
```

Comment:
@torkelo any reason this may be a bad idea?  The issue I have is that backend plugins do not have access to uid :( and we need something to pass forward that is consistent.

Long term, we should likely replace all use of `id` with `uid`, this change can help with that migration -- at least I do not think it will hurt.

Reply:
dam I thought the backend was already using uid

Reply:
When/where do you need to lookup ds by id?

Reply:
The channel response from a query is:
`ds/${id}/path`

Ideally that could be: `ds/${uid}/path`, BUT backend plugin API does not pass in the UID so it can not be returned from the query easily.  This is something we should fix, but is a bigger issue.

For V8, perhaps we should stop returning the internal id entirely?

Reply:
discussed with @torkelo -- he says it is an OK temporary fix until we get #32853